File: cpp/14-longest-common-prefix/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        std::sort(strs.begin(), strs.end());
        std::string result = "";
        std::string str1 = strs[0];
        std::string str2 = strs[strs.size() - 1];

        for (int i = 0; i < str1.size(); i++) {
            if (str1[i] == str2[i]) {
                result += str1[i];
            }
            else {
                break;
            }
        }
        return result;
    }
};

File: java/14-longest-common-prefix/solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int index = 0;
        Arrays.sort(strs);
        String str1 = strs[0];
        String str2 = strs[strs.length - 1];

        while (index < str1.length() && index < str2.length()) {
            if (str1.charAt(index) == str2.charAt(index)) {
                index++;
            }
            else {
                break;
            }
        }
        return str1.substring(0, index);
    }
}
